services/datagrid-service.json

```diff
@@ -0,0 +1,442 @@
+                  "intervalSeconds": 20,
+                  "timeoutSeconds": 1200,
+                  "maxUnavailable": 1,
+                  "maxSurge": 1
+               }
+            },
+            "template": {
+               "metadata": {
+                  "labels": {
+                     "application": "${APPLICATION_NAME}",
+                     "deploymentConfig": "${APPLICATION_NAME}"
+                  },
+                  "name": "${APPLICATION_NAME}"
+               },
+               "spec": {
+                  "containers": [
+                     {
+                        "env": [
+                           {
+                              "name": "USE_FIXED_MEMORY_SIZE",
+                              "value": "true"
+                           },
+                           {
+                              "name": "JGROUPS_PING_PROTOCOL",
+                              "value": "openshift.DNS_PING"
+                           },
+                           {
+                              "name": "OPENSHIFT_DNS_PING_SERVICE_NAME",
+                              "value": "${APPLICATION_NAME}-ping"
+                           },
+                           {
+                              "name": "OPENSHIFT_DNS_PING_SERVICE_PORT",
+                              "value": "8888"
+                           },
+                           {
+                              "name": "HTTPS_KEYSTORE",
+                              "value": "${APPLICATION_NAME}.p12"
+                           },
+                           {
+                              "name": "HTTPS_KEYSTORE_DIR",
+                              "value": "/var/run/secrets/java.io/keystores"
+                           },
+                           {
+                              "name": "HTTPS_NAME",
+                              "value": "NOT USED"
+                           },
+                           {
+                              "name": "HTTPS_PASSWORD",
+                              "value": "${APPLICATION_NAME}"
+                           },
+                           {
+                              "name": "SASL_SERVER_NAME",
+                              "value": "datagrid-service"
+                           },
+                           {
+                              "name": "INFINISPAN_CONNECTORS",
+                              "value": "hotrod,rest"
+                           },
+                           {
+                              "name": "HOTROD_AUTHENTICATION",
+                              "value": "TRUE"
+                           },
+                           {
+                              "name": "HOTROD_ENCRYPTION",
+                              "value": "TRUE"
+                           },
+                           {
+                              "name": "REST_SECURITY_DOMAIN",
+                              "value": "ApplicationRealm"
+                           },
```

Comment:
Most of these env variables should be set in `infinispan_service_profile.sh`, with the template simplified to something like:

```
"env": [
                           {
                              "name": "SERVICE_NAME",
                              "value": "${APPLICATION_NAME}"
                           },
                           {
                              "name": "SERVICE_PROFILE",
                              "value": "datagrid-service"
                           },
                           {
                              "name": "JGROUPS_PING_PROTOCOL",
                              "value": "openshift.DNS_PING"
                           },
                           {
                              "name": "OPENSHIFT_DNS_PING_SERVICE_NAME",
                              "value": "${APPLICATION_NAME}-ping"
                           },
                           {
                              "name": "USERNAME",
                              "valueFrom": {
                                 "secretKeyRef" : {
                                    "name" : "${APPLICATION_NAME}",
                                    "key" : "application-user"
                                 }
                              }
                           },
                           {
                              "name": "PASSWORD",
                              "valueFrom": {
                                 "secretKeyRef" : {
                                    "name" : "${APPLICATION_NAME}",
                                    "key" : "application-password"
                                 }
                              }
                           }
                        ],
```